packages/compile-solidity/src/compilerSupplier/Cache.ts

```diff
@@ -24,11 +26,30 @@ export class Cache {
   add(code: string, fileName: string) {
     const filePath = this.resolve(fileName);
     fs.writeFileSync(filePath, code);
+    this.memoizedCompilers[filePath] = code;
   }
 
   has(fileName: string) {
-    const file = this.resolve(fileName);
-    return fs.existsSync(file);
+    const filePath = this.resolve(fileName);
+    return fs.existsSync(filePath);
+  }
+
+  loadFile(fileName: string): string {
+    const filePath = this.resolve(fileName);
+    if (this.memoizedCompilers.has(filePath)) {
+      return this.memoizedCompilers[filePath];
+    }
+    try {
+      const compiler = fs.readFileSync(filePath).toString();
+      this.memoizedCompilers[filePath] = compiler;
+      return compiler;
+    } catch (error) {
+      if (!error.message.includes("ENOENT: no such file")) {
+        throw error;
+      } else {
+        throw new Error("The file specified could not be found.");
```

Comment:
Would this be a case for `@truffle/error`?

Reply:
Hmm probably not, right? This would most likely not be directly exposed to a user.